### PR TITLE
Remove deprecated generator flag from kubectl autoscale

### PR DESCRIPTION
**Pull request by MadhavJivrajani**
Signed-off-by: Madhav Jivrajani <madhav.jiv@gmail.com>

#### What type of PR is this?
/kind cleanup
/kind deprecation

#### What this PR does / why we need it:
This PR removes the `generator` deprecated flag in `staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale.go`

#### Which issue(s) this PR fixes:

Partial fix for #99897

#### Does this PR introduce a user-facing change?

```release-note
Remove deprecated --generator flag from kubectl autoscale
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```
/assign @rikatz 
/cc @rikatz

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale.go:121]: /hold
that's too soon, the flag was deprecated in https://github.com/kubernetes/kubernetes/pull/92998 even though it's not used and not functional [deprecation policy](https://kubernetes.io/docs/reference/using-api/deprecation-policy/) clearly states the time line and we need to keep it.

**Review by soltysh — CHANGES_REQUESTED** [review]

**Comment by rikatz** [comment]
ouch my bad. Though that with the full removal this could also be removed as well :/

**Comment by rikatz** [comment]
Does this part of the main issue (In staging/src/k8s.io/kubectl/pkg/cmd/create/create_deployment.go the call to AddGeneratorFlag must be replaced to cmdutil.AddDryRunFlag(cmd) and then in staging/src/k8s.io/kubectl/pkg/cmd/util/helpers.go the function AddGeneratorFlags can be removed) apply as well?

File: staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale.go
Comment (rikatz) [reply staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale.go:121]: So per deprecation policy (sorry, I've missed that when opening the issue!)
```
Rule #5b: CLI elements of admin-facing components (e.g. kubelet) must function after their announced deprecation for no less than:
    GA: 6 months or 1 release (whichever is longer)
```

As that PR was merged for v1.20 (which was released on Dec/2020) we should wait until jun/2020 to get this merged right?

Probably the timeline for v1.22 (that's going to be released in...Jun/2020 I guess) would allow us to merge this, so the good approach here is to wait until v1.21 is released and then merge this, is this right?

Thanks @soltysh

Comment (pacoxu) [reply staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale.go:121]: 1.21 will release on April，2021.
need we add milestone 1.22 for this PR?
/milestone 1.22

Comment (rikatz) [reply staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale.go:121]: @pacoxu we are almost at code freeze anyway, so we can wait :)

**Comment by rikatz** [comment]
/retitle Remove deprecated generator flag from kubectl autoscale

**Review by soltysh — APPROVED** [review]
/lgtm
/approve
/hold cancel
/retest

**Pull request merged**
